File: 2026/April/LC-2515-Shortest-Distance-to-Target-String-in-a-Circular-Array/solution.py
class Solution:
    def closestTarget(self, words: List[str], target: str, startIndex: int) -> int:
        n = len(words)
        if words[startIndex] == target: return 0
        ans = n + 1
        for i in range(1, (n // 2) + 1):
            if words[(startIndex - i + n) % n] == target:
                ans = min(ans, i)
            if words[(startIndex + i) % n] == target:
                ans = min(ans, i)
        return ans if ans != n + 1 else -1

File: 2026/April/LC-2515-Shortest-Distance-to-Target-String-in-a-Circular-Array/solution.cpp
class Solution {
    public:
        int closestTarget(vector<string>& words, string target, int startIndex) {
            int n = words.size();
            if (words[startIndex] == target) return 0;
            int ans = n + 1;
            for (int i = 1; i <= n / 2; i++) {
                if ((words[(startIndex - i + n) % n]) == target) {
                    ans = min(ans, i);
                } else if ((words[(startIndex + i) % n]) == target) {
                    ans = min(ans, i);
                }
            }
            return (ans == n + 1) ? -1 : ans;
        }
    };